src/models/hooks/activityReport.js

```diff
@@ -190,8 +190,11 @@ const determineObjectiveStatus = async (activityReportId, sequelize, isUnlocked)
   const objectiveIds = objectives.map((o) => o.objectiveId);
 
   // Get all the reports that use the objectives
-  const allObjectiveReports = await sequelize.models.ActivityReport.findAll({
+  const approvedReports = await sequelize.models.ActivityReport.findAll({
     attributes: ['id', 'calculatedStatus', 'endDate'],
+    where: {
+      calculatedStatus: REPORT_STATUSES.APPROVED,
+    },
```

Comment:
This restriction on results was not done on prepose to be able to reduce the number of queries t the database. as each request is slow relative to minor post filtering on the within the hook's js